tests/scalers/neo4j/neo4j_test.go

```diff
@@ -0,0 +1,299 @@
+//go:build e2e
+// +build e2e
+
+package neo4j_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/joho/godotenv"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	. "github.com/kedacore/keda/v2/tests/helper"
+)
+
+// Load environment variables from .env file
+var _ = godotenv.Load("../../.env")
+
+const (
+	testName = "neo4j-test"
+)
+
+var (
+	testNamespace    = fmt.Sprintf("%s-ns", testName)
+	deploymentName   = fmt.Sprintf("%s-deployment", testName)
+	secretName       = fmt.Sprintf("%s-secret", testName)
+	triggerAuthName  = fmt.Sprintf("%s-ta", testName)
+	scaledObjectName = fmt.Sprintf("%s-sobj", testName)
+	neo4jNamespace   = "neo4j-ns"
+	neo4jUser        = "neo4j"
+	neo4jHelmRepo    = "https://helm.neo4j.com/neo4j"
+	minReplicaCount  = 1
+	maxReplicaCount  = 2
+)
+
+type templateData struct {
+	TestNamespace                string
+	DeploymentName               string
+	HostName                     string
+	Port                         string
+	Username                     string
+	Password                     string
+	Neo4jNamespace               string
+	SecretName                   string
+	TriggerAuthName              string
+	ScaledObjectName             string
+	Connection, Base64Connection string
+}
+
+const (
+	deploymentTemplate = `
```

Comment:
Why are we leaving the first line of the manifest empty?

Reply:
No specific reason, the yaml template was taken correctly that way by the `kubectlApplyTemplate` function